godot-core/src/builtin/rect2i.rs

```diff
@@ -78,21 +82,492 @@ impl Rect2i {
         self.size = end - self.position
     }
 
-    /* Add in when `Rect2i::abs()` is implemented.
+    /// Returns a `Rect2i` with equivalent position and area, modified so that the top-left corner
+    /// is the origin and `width` and `height` are positive.
+    #[inline]
+    pub fn abs(self) -> Self {
+        let abs_size = self.size.abs();
+        let offset = Vector2i::new(cmp::min(self.size.x, 0), cmp::min(self.size.y, 0));
+        Self::new(self.position + offset, abs_size)
+    }
+
+    /// Returns `true` if this `Rect2i` completely encloses another one.
+    #[inline]
+    pub const fn encloses(&self, other: Self) -> bool {
+        self.assert_nonnegative();
+        other.assert_nonnegative();
+        // FIXME: provide an unsafe version without the check?
+        // FIXME: or just remove the check altogether?
+        let own_end = self.end();
+        let other_end = other.end();
+        other.position.x >= self.position.x
+            && other.position.y >= self.position.y
+            && other_end.x <= own_end.x
+            && other_end.y <= own_end.y
+    }
+
+    /// Returns a copy of this `Rect2i` expanded so that the borders align with the given point.
+    #[inline]
+    pub fn expand(self, to: Vector2i) -> Self {
+        self.assert_nonnegative();
+        // FIXME: provide an unsafe version without the check?
+        // FIXME: or just remove the check altogether?
+        let mut end = self.end();
+        let mut begin = self.position;
+
+        begin.x = cmp::min(begin.x, to.x);
+        begin.y = cmp::min(begin.y, to.y);
+        end.x = cmp::max(end.x, to.x);
+        end.y = cmp::max(end.y, to.y);
+
+        Self::from_corners(begin, end)
+    }
+
+    /// Returns the area of the `Rect2i`.
+    #[inline]
+    pub const fn get_area(&self) -> i32 {
+        self.size.x * self.size.y
+    }
```

Comment:
In Rust, getters typically don't have the `get_` prefix, unless there's ambiguity.

I would just call it `area`. Same for `center`.